### PR TITLE
Always resolve messages

### PR DESCRIPTION
**Pull request by lennym**
At the moment failed messages stay on the queue and are periodically re-processed. This is bad.

It means that the client will have already erroed out, and will likely attempt to resend the same or similar message. This will result in duplicate messages on the queue, and so in all likelihood, more errors.

This in turn makes a mess of the logs, and can potentially have unexpected side effects if for whatever reason a previously failed message becomes deliverable.

The best thing to do is to discard failed messages (by treating them as processed) but by virtue of not having written to the changelog, the client will throw.

tl;dr - it's better to know the thing hasn't been done if it errors, than for it to maybe happen at some unknown point in the future.

### CODE REVIEW

**Comment by wheelsandcogs** [comment]
Another option would be to use a dead-letter queue which is for exactly this scenario:
https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html

...but that's only worth doing if the dead messages actually get looked at at some point in the future, even if it's only for debug - if there's no one monitoring the dead letter queue then you might as well just bin them off immediately.

**Comment by lennym** [comment]
Yeah, I'm aware of that concept, but I'm not 100% clear on what the benefits are. It seems to me that it involves setting up a raft of extra infrastructure - an extra message queue, and then another worker to consumer it - to just do what we're already doing here, namely to push the failure to logs

Admittedly, there's a gap in our logging and alerting at the moment, which needs to be filled.
